session_test.go

```diff
@@ -405,20 +406,19 @@ func TestNegotiateStreamError(t *testing.T) {
 	defer cancel()
 	clientConn, serverConn := net.Pipe()
 	clientJID := jid.MustParse("me@example.net")
-	go func() {
-		s, err := xmpp.ReceiveSession(ctx, serverConn, 0, xmpp.NewNegotiator(xmpp.StreamConfig{
+	var wg sync.WaitGroup
```

Comment:
I think a wait group is the wrong synchronization mechanism to use here. We'll never add more goroutines, so the WG will always just be set to 1. This isn't a huge problem, but it means we're importing `sync` when we don't really need it and using something with semantics that make it seem like there will be more goroutines when we're reading the code later.

Instead I think a `chan struct{}` that gets closed at the end of the goroutine (and then received from at the end of the test) would be a better solution that doesn't have the added baggage of importing `sync`. Alternatively, we could use a `chan error` that sends the error back into the main goroutine for logging. I'm not sure which is better, but I lean towards the `chan struct{}` no idea why though.

This also has the added issue that if the call to `ReceiveSession` never returns for some reason the test will deadlock. I'm not sure what to do about that; we could put a timeout on the context, but that will make the tests potentially flaky and I never think it's a good idea. That may be all we can do though if we want to log the output from ReceiveSession.

Reply:
Applied the changes as you suggested 👍